File: code/rent-exemption/rent-exemption.en.ts
import {
    Connection,
    clusterApiUrl,
} from "@solana/web3.js";

(async () => {
    const connection = new Connection(clusterApiUrl('devnet'), 'confirmed');

    // length of data in the account to calculate rent for
    const dataLength = 1500;
    const rentExemptionAmount = await connection.getMinimumBalanceForRentExemption(dataLength);
    console.log({
        rentExemptionAmount
    })
})();

File: code/accounts/rent-exemption/rent-exemption.en.rs
use solana_client::rpc_client::RpcClient;
use solana_sdk::commitment_config::CommitmentConfig;

fn main() {
    let rpc_url = String::from("https://api.devnet.solana.com");
    let connection = RpcClient::new_with_commitment(rpc_url, CommitmentConfig::confirmed());
    let data_length = 1500;

    let rent_exemption_amount = connection
        .get_minimum_balance_for_rent_exemption(data_length)
        .unwrap();

    println!("rent exemption amount: {}", rent_exemption_amount);
}
